src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -1566,16 +1567,53 @@ export class HtmlVideoPlayer {
             return t.Index === streamIndex;
         })[0];
 
-        this.setTrackForDisplay(this.#mediaElement, track, targetTextTrackIndex);
-        if (enableNativeTrackSupport(this._currentPlayOptions?.mediaSource, track)) {
-            if (streamIndex !== -1) {
-                this.setCueAppearance();
-            }
+        // This play method can only check if it is real direct play, and will mark Remux as Transcode as well
+        const isDirectPlay = this._currentPlayOptions.playMethod === 'DirectPlay';
+        const burnInWhenTranscoding = appSettings.alwaysBurnInSubtitleWhenTranscoding();
+
+        let sessionPromise;
+        if (!isDirectPlay && burnInWhenTranscoding) {
+            const apiClient = ServerConnections.getApiClient(this._currentPlayOptions.item.ServerId);
+            sessionPromise = apiClient.getSessions({
+                deviceId: apiClient.deviceId()
+            }).then(function (sessions) {
+                return sessions[0] || {};
+            }, function () {
+                return Promise.resolve({});
+            });
         } else {
-            // null these out to disable the player's native display (handled below)
-            streamIndex = -1;
-            track = null;
+            sessionPromise = Promise.resolve({});
         }
+
+        const player = this;
+
+        sessionPromise.then((s) => {
+            if (!s.TranscodingInfo || s.TranscodingInfo?.IsVideoDirect) {
```

Comment:
Unnecessary optional chaining.
```suggestion
            if (!s.TranscodingInfo || s.TranscodingInfo.IsVideoDirect) {
```